qiskit/providers/aer/aerprovider.py

```diff
@@ -26,7 +27,8 @@ def __init__(self, *args, **kwargs):
         # Populate the list of Aer simulator providers.
         self._backends = [QasmSimulator(provider=self),
                           StatevectorSimulator(provider=self),
-                          UnitarySimulator(provider=self)]
+                          UnitarySimulator(provider=self),
+                          CHSimulator(provider=self)]
```

Comment:
Yeah, I think that would work. Most of the CHController is a shadow of the QasmController API anyway.

The one addition I made to the API is the required memory validation. We could add something like this to the QasmController? Or else I port it to Python, but either way we'd need to update the _validate method of QasmSimulator to account for the different footprints of the CH/Statevector methods.

Reply:
Adding the memory validation to the QasmController sounds good. We've been meaning to do something like this for awhile, since we should really be doing it for the statevector simulator as well (stabilizer too though in that case its memory footprint is comparatively tiny).

Currently the memory is checked in python before sending the qobj so it can fail-fast, but if we handle it correctly in the C++ we could remove that part from the python backend (or just do a very basic check there).

Reply:
For the moment I'll leave the simple check in Python, and set it to auto-switch to ch in the event that the number of qubits is too much for the statevector method (as long as it's not too high for the ch method). I guess the thing to do would be to make use of the `required_memory_mb` method of state as part of an initialisation step in QasmController?